### PR TITLE
Switch to directory-based patch management

### PR DESCRIPTION
**Pull request by nathanchance**
This should make updating patches a little bit easier. I still envision the workflow being something like this:

1. Apply patches to Linux repo.
```
$ git am <patches_folder>/*.patch
```
2. Drop or modify patches.
3. Remove current patch set.
```
$ rm <patches_folder>/*.patch
```
4. Generate current patch set and update series file
```
$ git format-patch --base=<whatever> -o <patches_folder> <whatever>.. | sed "s;<patches_folder>/;;g" > <patches_folder>/series
```

Example on the last command:

```
$ git format-patch --base=v5.15-rc2 -o $CBL_GIT/ci/patches/mainline/ v5.15-rc2.. | sed "s;$CBL_GIT/ci/patches/mainline/;;g" > $CBL_GIT/ci/patches/mainline/series
```

I can document this later if it would be helpful. I am also open to exploring other models for managing patches.

### CODE REVIEW

File: generate_workflow.py [hunk 1]
```diff
@@ -169,9 +170,11 @@ def get_steps(build, build_set):
...
 def patch_series(tree_name):
     ci_folder = pathlib.Path(__file__).resolve().parent
-    mbox = ci_folder.joinpath("patches", "%s.mbox" % tree_name)
-    mbox_rel_path = mbox.relative_to(ci_folder).as_posix()
-    return " --patch-series {}".format(mbox_rel_path) if mbox.exists() else ""
+    patches_folder = pathlib.Path("patches").joinpath(tree_name)
+    patch_files = glob.glob(
+        ci_folder.joinpath(patches_folder).as_posix() + "/*.patch")
+    return " --patch-series {}".format(
+        patches_folder.as_posix()) if patch_files else ""
```
Comment (nickdesaulniers) [inline generate_workflow.py:177]: there's duplication between the additions to generate_workflow.py and generate_tuxsuite.py. Can you move this to utils.py and try to reuse more code between the two?

File: check-patches.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    # First, make sure series file is not missing
+    if [[ ! -f ${series} ]]; then
+        echo "${folder} exists but ${series} doesn't?"
+        echo
+        echo "Generate it with the following commands:"
+        update_series_commands
+        exit 1
+    fi
+
+    # Next, check that all of the patches in the series file exist (removed a patch, did not update series file)
```
Comment (nickdesaulniers) [inline check-patches.sh:26]: is this comment < 80 lines?

File: check-patches.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        if [[ ! -f ${folder}/${patch} ]]; then
+            echo "${folder} does not contain ${patch} but it is in ${series}?"
+            echo
+            echo "Update the series file:"
+            update_series_commands
+            exit 1
+        fi
+    done <"${series}"
+
+    # Lastly, make sure that all of the patches in the patches folder are in the series file (removed from series file, did not remove patch file)
```
Comment (nickdesaulniers) [inline check-patches.sh:37]: wrap comment?

**Review by nickdesaulniers — COMMENTED** [review]
thanks for the patches! 😵‍💫 🦘

**Comment by nickdesaulniers** [comment]
can you rebase on #202 to get the mulodi4 patch that I'm about to merge?

**Comment by nathanchance** [comment]
Yes, I will address your comments and rebase so that I can drop the hack patch that I added (as mainline is not going to actually test your patch because we won't build the NBD code).

**Comment by nathanchance** [comment]
Alright, I addressed all of your comments and rebased on #202, in addition to adding patches for most of the issues for trees based on -next or -tip. PTAL.

**Review by nickdesaulniers — APPROVED** [review]
looks great! nice work! 🟧 🟨

**Comment by nathanchance** [comment]
Thanks for the review!

**Pull request merged**